source/extensions/filters/http/health_check/health_check.cc

```diff
@@ -104,7 +104,11 @@ Http::FilterHeadersStatus HealthCheckFilter::encodeHeaders(Http::ResponseHeaderM
     }
 
     headers.setEnvoyUpstreamHealthCheckedCluster(context_.localInfo().clusterName());
-  } else if (context_.healthCheckFailed()) {
+  }
+
+  if (context_.healthCheckFailed() &&
+      Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.health_check.immediate_failure_exclude_from_cluster")) {
```

Comment:
This ends up controlling whether the upstream sets the header right? Based on the name I would have expected it to be on the consuming side instead. Maybe clarify so that people would know whether the runtime should be set for the upstream or downstream Envoy instances?

Reply:
Yeah I just used the same flag to control both behaviors. Do you think I should split the flags?

Reply:
I made the release note more clear, let me know what you think.

Reply:
Yeah that's what I was thinking, just making it clear where the flag needs to be set